### PR TITLE
(feat) Group Session Visits

### PR DESCRIPTION
**Pull request by ZacButko**
## Requirements

- [x] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [x] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide).
- [ ] My work includes tests or is validated by existing tests.

## Summary
This includes many changes but the most important change is that this is now saving a visit for a user when entering the Group Session data. After the visit is created the form data is posted with the new visitUUID attached. In then end when we check the patient's chart we can see their data now shows up on the visit.

## Screenshots
https://user-images.githubusercontent.com/5445264/194053739-38a40780-8a73-4e4f-9c26-1380b061dfad.mov

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -53,7 +53,7 @@
   },
   "devDependencies": {
     "@carbon/react": "^1.9.0",
-    "@openmrs/esm-framework": "next",
```
Comment (vasharma05) [inline package.json:56]: Shouldn't this be kept for local purpose only?

File: src/context/FormWorkflowReducer.ts [hunk 2]
```diff
@@ -259,6 +259,8 @@ const reducer = (state, action) => {
         activeFormUuid: null,
       };
       persistData(newState);
+      //eslint-disable-next-line
```
Comment (vasharma05) [inline src/context/FormWorkflowReducer.ts:262]: Why using eslint-disable?

**Review by hadijahkyampeire — APPROVED** [review]
LGTM, just a minor suggestion, wouldn't it be better to make one `ComposeModal` Component that can be reused for the `Cancel` and `Complete` I see the content is almost the same.

**Review by vasharma05 — APPROVED** [review]
LGTM!

File: package.json (hunk 1 above)
Comment (hadijahkyampeire) [reply package.json:56]: I thought so as well but maybe we need to push the updated version if it won't break stuff, I think everyone needs to point to the right version of `openmrs-framework`. I will defer to Zac on what is best.

File: src/context/FormWorkflowReducer.ts (hunk 2 above)
Comment (ZacButko) [reply src/context/FormWorkflowReducer.ts:262]: The next line includes a template literal variable syntax in a non template literal string. This throws an eslint error, so the eslint-disable-next-line suppresses it.

**Comment by ZacButko** [comment]
> LGTM, just a minor suggestion, wouldn't it be better to make one `ComposeModal` Component that can be reused for the `Cancel` and `Complete` I see the content is almost the same.

The components are sufficiently different that it benefits having two different functions. One has two buttons, the other has three. Each modal calls very specific functions from the context it operates in, really if you were to try to abstract this you would be replicate most logic with two results, so in the end it is much more clear just to have two different components

**Pull request merged**